test/recipes/70-test_tls13messages.t

```diff
@@ -296,4 +301,61 @@ checkhandshake($proxy, checkhandshake::HRR_RESUME_HANDSHAKE,
                | checkhandshake::PSK_CLI_EXTENSION
                | checkhandshake::PSK_SRV_EXTENSION,
                "Resumption handshake with HRR test");
+
+#Test 16: signature_algorithms with 1.3-only ClientHello
+my $testtype = PURE_SIGALGS;
+my $dsa_status = 1;
+$proxy->clear();
+$proxy->clientflags("-tls1_3");
+$proxy->filter(\&modify_sigalgs_filter);
+$proxy->start();
+ok($dsa_status, "DSA/SHA2 sigalg sent for 1.3-only ClientHello");
+
+#Test 17: signature_algorithms with backwards compatible ClientHello
+$testtype = COMPAT_SIGALGS;
+$dsa_status = 1;
```

Comment:
Perhaps it is better to initialise to 0 and have the test set it to 1 on success. Otherwise if something fundamental goes wrong and the filter doesn't get invoked then the test still passes.